doc/user/content/overview/architecture.md

```diff
@@ -8,23 +8,124 @@ menu:
+
+[todo: operational concerns for clusters].
+todo: you want to have more indexes on a given cluster to enable more sharing/reusing of data in all of the cluster replicas
+todo: also the more indexes and materialized views you have in a given cluster potentially its cheaper for you
+todo: but you want to have fewer indexes per cluster to decrease the chances that one slow index affects other indexes on the same cluster
+todo: you might also want different clusters to account for organizational boundaries
+todo: or you might put similar workloads (like a cluster full of mat views that other clusters read from) in the same cluster
+todo: you can spin up a new cluster to migrate to new version ... of materialize? but does that even make sense now?
+todo: what happens if you want to change a view and change the business logic you are performing? iiuc clusters and cluster replicas don't help much with that?
+
+#### Cluster replicas
+
+Where clusters represent the logical set of computations you want to maintain, cluster replicas are their physical counterparts. Cluster replicas are where Materialize actually creates and maintains computations.
+
+Each cluster replica is essentially a clone, constructing the same indexes and materialized views. [TODO: is this also true of ad-hoc select queries?]. Each cluster replica receives a copy of all data that comes in from sources its dataflows use, and uses the data to perform identical computations. This design provides Materialize with active replication, and so long as one replica is still reachable, the cluster continues making progress.
+
+[todo: operational concerns for cluster replicas]
+todo: you can add more replicas (to be more crash tolerant)
+todo: you can scale replica sizes up and down (to be more performant)
+todo: are there weird wrinkles here? like if I have 1 xl replica and one small replica my expectation is that i will get the speed of the xl with the fault tolerance associated with having a second replica, or are there cases where my speed will be bottlenecked by the small?
+
+#### Indexes
+
+Indexes assemble and maintain a a source, unmaterialized or materialized view's data in memory within all cluster replicas associated with a given cluster. Users can create indexes on any source, unmaterialized view, or materialized view present in the region. This enables future `SELECT` statements against that source, unmaterialized or materialized view to complete much faster, because the data can be directly read from in memory from a cluster replica rather than needing to be read from storage.
```

Comment:
```suggestion
Indexes assemble and maintain a source, views or materialized view's data in memory within all cluster replicas associated with a given cluster. Users can create indexes on any source, view, or materialized view present in the region. This enables future `SELECT` statements against that source, unmaterialized or materialized view to complete much faster, because the data can be directly read from in memory from a cluster replica rather than needing to be read from storage.
```